core/src/main/java/io/aiven/klaw/repository/AclRequestsRepo.java

```diff
@@ -7,8 +7,10 @@
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.data.repository.query.Param;
+import org.springframework.data.repository.query.QueryByExampleExecutor;
 
-public interface AclRequestsRepo extends CrudRepository<AclRequests, AclRequestID> {
+public interface AclRequestsRepo
+    extends CrudRepository<AclRequests, AclRequestID>, QueryByExampleExecutor<AclRequests> {
```

Comment:
I hope we are good by using QueryByExampleExecutor

Query by Example API has some limitations. For instance:

Nesting and grouping statements are not supported, for example:  (firstName = ?0 and lastName = ?1) or seatNumber = ?2
String matching only includes exact, case-insensitive, starts, ends, contains, and regex
All types other than String are exact-match only

Reply:
We still require @Query for more complex operation, Query by example is only to be used when we are looking for exact matches.

e.g. If the User were to send in a partial topic name we would need a different type of API to be exposed and used imo as it would require probably custom queries.
QueryByExample here should only be when we are filtering using the dropdowns and other data we know fully.